apps/vaporgui/VStringLineEdit.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <iostream>
+
+#include <string>
+#include <sstream>
+#include <iomanip>
+
+#include "VContainer.h"
+
+class QStringLineEdit;
+
+//! \class VStringLineEdit
+//! \ingroup Public_GUI
+//! \brief A wrapper for a QLineEdit that handles user input of type string, 
+//! and provides Vapor's standard setters, getters, and signals
+
+class VStringLineEdit : public VContainer {
+    Q_OBJECT
+
+    public:
+        VStringLineEdit( std::string value = "" );
+
+        //! Set the current string value in the line edit
+        virtual void SetValueString( std::string value );
+   
+        //! Get the current value in the line edit 
+        virtual std::string GetValueString() const;
+
+    protected:
+        QLineEdit*  _lineEdit;
```

Comment:
VStringEdit::_lineEdit should be private because of the idea it VStringEdit is the single access-point to QLineEdit and other classes shouldn’t be able to modify it as the children of VStringEdit do.

Reply:
Agreed.